### PR TITLE
[Release] Update release note location

### PR DESCRIPTION
**Pull request by itsjustkevin**
**TLDR:** Move the changelog, currently called ["hotfixes to the stable channel"](https://github.com/flutter/flutter/blob/master/docs/releases/Hotfixes-to-the-Stable-Channel.md), to the root directory of Flutter/Flutter and rename it to CHANGELOG.md before the Q3 release cutoff.

This PR accomplishes the following:
1. Renames 'Hotfixes-to-the-stable-channel' to CHANGELOG.md
2. Moves CHANGELOG.md to Flutter/Flutter root.
3. Update references to 'Hotfixes-to-the-stable-channel' to CHANGELOG.md.

---

#### Background
Flutter has historically kept documentation on wiki pages in the Flutter repository. Recently, we executed a large-scale migration of our wiki into a `docs` folder within flutter/flutter.

#### Proposal
I propose relocating the current changelog, which is called ["hotfixes to the stable channel"](https://github.com/flutter/flutter/blob/master/docs/releases/Hotfixes-to-the-Stable-Channel.md), to the root directory of the Flutter/Flutter repository and renaming it to CHANGELOG.md. This change aims to improve the visibility, accessibility, and consistency of our documentation processes.

#### Benefits
1. **Unify Dart and Flutter Processes:** This move aligns Flutter’s changelog management with Dart's practices. As the release manager for both Dart and Flutter, this unification will streamline the processes and ensure consistency across projects.

#### Timeline
I would like this proposal **approved or denied before the Q3 release cutoff** so it may be available within the stable branch.

---

> [!NOTE] 
> This proposal aims to improve our workflow and documentation standards. I am open to feedback and further discussion on how we can best implement this change.

### CODE REVIEW

**Review by itsjustkevin — COMMENTED** [review]
Update of: https://github.com/flutter/flutter/pull/149815

**Comment by matanlurey** [comment]
test-exempt: Changing links (strings) in a way that is expected.

**Review by Piinks — APPROVED** [review]
LGTM!

**Comment by itsjustkevin** [comment]
My initial plan was to have this change in before the Q3 release cutoff to ensure that it is included in the next stable.  Should we let it show up in the next stable release or cherry-pick into this release?

Thoughts @christopherfujino?

**Comment by christopherfujino** [comment]
> My initial plan was to have this change in before the Q3 release cutoff to ensure that it is included in the next stable. Should we let it show up in the next stable release or cherry-pick into this release?
> 
> Thoughts @christopherfujino?

I don't have a strong opinion either way.
